velox/functions/sparksql/String.h

```diff
@@ -1556,4 +1556,73 @@ struct Empty2NullFunction {
   }
 };
 
+template <typename TOutString, typename TInString>
+FOLLY_ALWAYS_INLINE void trimTrailingSpaces(
+    TOutString& output,
+    const TInString& inputStr,
+    int32_t numChars,
+    int32_t limit) {
+  auto numTailSpacesToTrim = numChars - limit;
+  VELOX_USER_CHECK_GT(numTailSpacesToTrim, 0);
+
+  auto curPos = inputStr.end() - 1;
+  auto trimTo = inputStr.end() - numTailSpacesToTrim;
+
+  while (curPos >= trimTo && stringImpl::isAsciiSpace(*curPos)) {
+    curPos--;
+  }
+  // Get num of chars.
+  auto trimmedSize = numChars - std::distance(curPos + 1, inputStr.end());
+
+  VELOX_USER_CHECK_LE(
+      trimmedSize,
+      limit,
+      "Exceeds char/varchar type length limitation: {}",
+      limit);
+  output.setNoCopy(
+      StringView(inputStr.data(), std::distance(inputStr.begin(), curPos + 1)));
+}
+
+template <typename T>
+struct VarcharTypeWriteSideCheckFunction {
```

Comment:
Can you add comments for this function to illustrate the semantics? Also consider moving the implementation to a separate file as the String.h is large.

Reply:
Maybe CharVarcharCodegenUtils.h

Reply:
I think the name 'CharVarcharCodegenUtils' could be reconsidered as this method is not relevant to codegen in Velox. Probably use `VarcharTypeWriteSideCheck.h` as name if only one helper function.